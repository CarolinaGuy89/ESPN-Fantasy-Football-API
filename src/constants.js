/**
 * Maps `slotCategoryId`'s numerical enum to readable positions.
 * @type {Object}
 */
const slotCategoryIdToPositionMap = {
  0: 'QB',
  1: 'TQB',
  2: 'RB',
  3: 'RB/WR',
  4: 'WR',
  5: 'WR/TE',
  6: 'TE',
  7: 'OP',
  8: 'DT',
  9: 'DE',
  10: 'LB',
  11: 'DL',
  12: 'CB',
  13: 'S',
  14: 'DB',
  15: 'DP',
  16: 'D/ST',
  17: 'K',
  18: 'P',
  19: 'HC',
  20: 'Bench',
  21: 'IR',
  22: 'Unknown?', // TODO: Figure out what this is
  23: 'RB/WR/TE',
  24: 'Unknown?' // TODO: Figure out what this is
};

export { slotCategoryIdToPositionMap };
