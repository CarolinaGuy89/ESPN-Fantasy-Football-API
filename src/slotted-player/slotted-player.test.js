import _ from 'lodash';

import BaseObject from '../base-object/base-object.js';
import Player from '../player/player.js';

import SlottedPlayer from './slotted-player.js';

import { localObject, serverResponse } from './slotted-player.stubs.js';

import { slotCategoryIdToPositionMap } from '../constants.js';

describe('SlottedPlayer', () => {
  test('extends BaseObject', () => {
    const slottedPlayer = new SlottedPlayer();
    expect(slottedPlayer).toBeInstanceOf(BaseObject);
  });

  describe('when creating a team from a server response', () => {
    test('parses and assigns data correctly', () => {
      const slottedPlayer = SlottedPlayer.buildFromServer(serverResponse);
      expect(slottedPlayer).toMatchSnapshot();
    });
  });

  describe('when creating a team locally', () => {
    test('parses and assigns data correctly', () => {
      const slottedPlayer = new SlottedPlayer(localObject);
      expect(slottedPlayer).toMatchSnapshot();
    });
  });


  describe('responseMap', () => {
    const buildSlottedPlayer = (data, options) => SlottedPlayer.buildFromServer(data, options);

    describe('player', () => {
      describe('manualParse', () => {
        describe('when the passed data is undefined', () => {
          test('sets undefined', () => {
            const slottedPlayer = buildSlottedPlayer({});
            expect(slottedPlayer.player).toBeUndefined();
          });
        });

        describe('when the passed data is empty', () => {
          test('sets undefined', () => {
            const data = { player: {} };

            const slottedPlayer = buildSlottedPlayer(data);
            expect(slottedPlayer.player).toBeUndefined();
          });
        });

        describe('when the passed data is populated', () => {
          test('sets a new player', () => {
            const playerId = 10;
            const data = {
              player: { playerId }
            };

            const slottedPlayer = buildSlottedPlayer(data);
            expect(slottedPlayer.player).toEqual(Player.buildFromServer({ playerId }));
          });
        });
      });
    });

    describe('position', () => {
      describe('manualParse', () => {
        test('sets the mapped position', () => {
          const position = 2;
          const data = { slotCategoryId: position };

          const slottedPlayer = buildSlottedPlayer(data);
          expect(slottedPlayer.position).toBe(_.get(slotCategoryIdToPositionMap, position));
        });
      });
    });

    describe('isLocked', () => {
      describe('manualParse', () => {
        describe('when valid enum key is passed', () => {
          test('switches on numerical enum correctly', () => {
            const lockedStatuses = {
              0: false,
              4: true
            };

            _.forEach(lockedStatuses, (value, key) => {
              const numKey = _.toNumber(key);
              const data = { lockStatus: numKey };

              const slottedPlayer = buildSlottedPlayer(data);
              expect(slottedPlayer.isLocked).toBe(value);
            });
          });
        });

        describe('when invalid enum key is passed', () => {
          test('sets error string', () => {
            const data = { lockStatus: 2 };

            const slottedPlayer = buildSlottedPlayer(data);
            expect(slottedPlayer.isLocked).toBeUndefined();
          });
        });
      });
    });
  });
});
