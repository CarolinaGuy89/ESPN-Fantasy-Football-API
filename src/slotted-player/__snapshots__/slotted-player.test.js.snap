// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlottedPlayer attribute population from local object parses data correctly 1`] = `
SlottedPlayer {
  "isKeeper": false,
  "isLocked": true,
  "player": Player {
    "eligiblePositions": Array [
      "QB",
      "Bench",
    ],
    "firstName": "Tom",
    "isActive": true,
    "isIREligible": false,
    "jerseyNumber": "12",
    "lastName": "Brady",
    "percentOwned": 99.80315,
    "percentOwnedChange": 0.20315,
    "percentStarted": 81.69291,
    "playerId": 2330,
  },
  "position": "QB",
}
`;

exports[`SlottedPlayer attribute population from server response parses data correctly 1`] = `
SlottedPlayer {
  "isKeeper": false,
  "isLocked": true,
  "player": Player {
    "eligiblePositions": Array [
      "QB",
      "Bench",
    ],
    "firstName": "Tom",
    "isActive": true,
    "isIREligible": false,
    "jerseyNumber": "12",
    "lastName": "Brady",
    "percentOwned": 99.80315,
    "percentOwnedChange": 0.20315,
    "percentStarted": 81.69291,
    "playerId": 2330,
  },
  "position": "QB",
}
`;
